### DISCUSSION TITLE
Urql with nextjs

### DISCUSSION THREAD

**Discussion by mooyg** [urql-graphql/urql#1767] [Q&A] [answered]
Hey, I am trying to use urql with next.js. 
```
export const getServerSideProps = wrapper.getServerSideProps(
  (store) => async (): Promise<GetServerSidePropsResult<any>> => {
    console.log("GET SERVER SIDE PROPS");
    const ssrCache = ssrExchange({ isClient: false });
    const client = initUrqlClient(
      {
        url: "http://localhost:4000/graphql",
        exchanges: [
          dedupExchange,
          cacheExchange,
          ssrCache,
          fetchExchange,
          devtoolsExchange,
        ],
        fetchOptions: () => {
          return {
            credentials: "include",
          };
        },
      },
      false
    );
    await client?.query(GET_USER).toPromise();

    return {
      props: {
        urqlState: ssrCache.extractData(),
      },
    };
  }
);
export default withUrqlClient(
  (ssr) => ({
    url: "http://localhost:4000/graphql",
  }),
  { ssr: false } // Important so we don't wrap our component in getInitialProps
)(LoginIndex);
```
currently having this I keep getting ```
PS C:\Users\mooy\Desktop\projects\locus> [0] (node:26228) UnhandledPromiseRejectionWarning: Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client``` basically I want to directly set the redux state from server side itself using `next-redux-wrapper` (currently not trying that cause I keep getting this error nor I can see the query in the devtools itself ) and not populate the cache ( I wanna know if this is possible too)

**Comment by mooyg** [comment, marked as answer]
nvm fixed it 😄

**Reply by adnanfuat** [reply]
is there any tutorial or document for ssg ?

**Reply by JoviDeCroock** [reply]
Are you looking for something that isn't covered in our docs? https://formidable.com/open-source/urql/docs/advanced/server-side-rendering/#using-getstaticprops-or-getserversideprops
